src/api/v1/users/password.ts

```diff
@@ -74,7 +75,18 @@ const routes: Hapi.ServerRoute[] = [
         payload: {
           token: Joi.string().length(64),
           password: passwordSchema,
-          confirmPassword: Joi.any().only(Joi.ref('password')),
+          passwordConfirm: Joi.string().required().valid(Joi.ref('password'))
+            .options({ language: { any: {
+              required: 'confirm password is required',
+              allowOnly: 'must match password',
+            } } }),
+        },
+        failAction: (request, h, err) => {
+          if ((<JoiBoomError> err).details[0].message
```

Comment:
IMO this is still a 400. The route has `auth: false`, anyone can access it, it just happens to require a token.

Reply:
The token is being used to authorise the password reset. We are sending back `400` if there is an error with a payload that is supplied by the user, it's validating their input. IMO `401` better suits token errors as it's authentication that we have created and send to the user in an email rather than the standard auth of a cookie.